src/crystal/system/thread.cr

```diff
@@ -1,40 +1,138 @@
+# :nodoc:
+module Crystal::System::Thread
+  # alias Handle
+
+  # def self.new_handle(thread_obj : ::Thread) : Handle
+
+  # def self.current_handle : Handle
+
+  # def self.yield_current : Nil
+
+  # def self.current_thread : ::Thread
+
+  # def self.current_thread=(thread : ::Thread)
+
+  # private def system_join : Exception?
+
+  # private def system_close
+
+  # private def stack_address : Void*
+end
+
+{% if flag?(:wasi) %}
+  require "./wasi/thread"
+{% elsif flag?(:unix) %}
+  require "./unix/pthread"
+{% elsif flag?(:win32) %}
+  require "./win32/thread"
+{% else %}
+  {% raise "Thread not supported" %}
+{% end %}
+
 # :nodoc:
 class Thread
+  include Crystal::System::Thread
+
+  # all thread objects, so the GC can see them (it doesn't scan thread locals)
+  protected class_getter(threads) { Thread::LinkedList(Thread).new }
+
+  @th : Crystal::System::Thread::Handle
```

Comment:
Maybe we could use this opportunity to change to a more explicit name like `@system_handle`?
(this could be refactored later, though)